### DISCUSSION TITLE
Cache Control not enabled in production

### DISCUSSION THREAD

**Discussion by kmanimurugan** [vercel/next.js#53287] [Help]
### Summary

In the self hosting
we can not able to see cache control in response header for the Images and Pages
Can you please help me resolve this?

**Comment by samuliasmala** [comment]
Same issue, when hosting in Vercel my main page (first html load) has proper `Cache-Control: public, max-age=0, must-revalidate` but the header is missing totally when self-hosting. Note that assets are hashed and cached properly also when self-hosting: `Cache-Control: public, max-age=31536000, immutable`

**Comment by leerob** [comment]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
